### PR TITLE
Test operator release

### PR DESCRIPTION
**Pull request by pminz**

### CODE REVIEW

File: .github/workflows/image_release.yaml [hunk 1]
```diff
@@ -0,0 +1,127 @@
+name: Gingersnap Operator - Build and publish
+
+on:
+  push:
+    branches:
+      - 'Branch Name'
+
+env:
+  GITHUB_TOKEN: ${{ secrets.PAT_TOKEN }}   
+  release-version: "0.0.0"
```
Comment (ryanemerson) [inline .github/workflows/image_release.yaml:10]: This needs to be set by an input. To minimise the changes to the rest of the action, we could doo:

`release-version: ${{ github.event.inputs.release }}`

File: .github/workflows/publish-main.yaml [hunk 2]
```diff
@@ -0,0 +1,75 @@
+name: Publish to main 
+
+on:
+  push:
```
Comment (ryanemerson) [inline .github/workflows/publish-main.yaml:4]: This approach won't work, as the action will only be triggered when a new commit is added to the specified branch. We require something that will be triggered by execution of the "release" action and continue to retry until a condition is satisfied.

Calling the [schedule-job-action](https://github.com/marketplace/actions/schedule-job-action) from the "release" action should allow this.

File: .github/workflows/publish-main.yaml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+      - name: Checkout community-operators repo
+        uses: actions/checkout@v2
+        with:
+          repository: k8s-operatorhub/community-operators
+          path: community-operators
+          ref: main
+          token: ${{ secrets.PAT_TOKEN }}
+
+      - name: Check if new version directory exists in community-operators repo
```
Comment (ryanemerson) [inline .github/workflows/publish-main.yaml:40]: Thinking about this more, I don't think this is sufficient as there's a delay between a community-operators PR being merged and OLM bundle image being available on OperatorHub.

Instead, we can use docker to check whether the required image has been published yet:

```
docker manifest inspect quay.io/operatorhubio/gingersnap:v${{ env.release-version}} > /dev/null
```

^ will return an exit code of 1 if the image doesn't exist yet, causing the step to fail.

File: .github/workflows/publish-main.yaml [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+        id: check_dir_1
+        run:  |
+          cd operator 
+          ./.github/scripts/directory.sh ${{ env.community_operators_dir_path}} ${{ env.release-version}}
+
+      - name: Checkout community-operators-prod repo
```
Comment (ryanemerson) [inline .github/workflows/publish-main.yaml:46]: There's no need for us to check `community-operators-prod` repo as we don't rely on this in the upstream CI.

File: .github/workflows/publish-main.yaml [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+          valueFile: 'config/manifests/bases/*-operator.clusterserviceversion.yaml'
+          branch: main
+          message: 'Next Version ${{ env.release-ersion }}' 
+          commitUserName: 'infinispan-qe-bot'
+          commitUserEmail: 'q*@infinispan.org'
+          token: ${{ secrets.PAT_TOKEN }}   
+          changes: |
+            {
+              "spec.version": "${{ env.release-version }}" ,
+              "metadata.name": "gingersnap-operator.v${{ env.release-version }}"
```
Comment (ryanemerson) [inline .github/workflows/publish-main.yaml:73]: There's no need to update the `metadata.name` or `version` field as `v0.0.0` is just a placeholder that is converted when we do `make bundle ...`

The field we must update is `replaces`, which should be set to the Operator version that has just been released.

**Review by ryanemerson — CHANGES_REQUESTED** [review]

File: .github/workflows/image_release.yaml [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+          push: true
+          tags: quay.io/gingersnap/operator:${{ env.release-version }}
+          build-args: |
+            OPERATOR_VERSION=${{ env.release-version }}
+
+      - name: Create OLM Bundle
+        run:  |
+            make operator-sdk
+            make bundle VERSION=${{ env.release-version }}
+            CHANNELS=alpha DEFAULT_CHANNEL=alpha IMG=quay.io/gingersnap/operator:${{ env.release-version }}
```
[the remark this answers is not in the document]
Comment (pminz) [reply .github/workflows/image_release.yaml]: updated in single line

File: .github/workflows/image_release.yaml [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+
+
+      - name: Checkout community-operators repo
+        uses: actions/checkout@v2
+        with:
+          repository: k8s-operatorhub/community-operators
+          path: community-operators
+          ref: main
+          token: ${{ secrets.PAT_TOKEN }}
```
[the remark this answers is not in the document]
Comment (pminz) [reply .github/workflows/image_release.yaml]: it was for testing purpose . removed

File: .github/workflows/image_release.yaml [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+
+
+      - name: Copy contents of bundle/ 
+        run: |
+          mkdir -p community-operators/operators/gingersnap/${{ env.release-version }}
+          cp -R bundle/* community-operators/operators/gingersnap/${{ env.release-version }}
+          cd community-operators
+          git config --list
```
[the remark this answers is not in the document]
Comment (pminz) [reply .github/workflows/image_release.yaml]: yes . removed

File: .github/workflows/image_release.yaml [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+        uses: peter-evans/create-pull-request@v4
+        with: 
+          token: ${{ secrets.PAT_TOKEN }}
+          path: community-operators
+          commit-message: 'Gingersnap Operator ${{ env.release-version }}'
+          committer: infinispan-qe-bot <q*@infinispan.org>
+          author: infinispan-qe-bot <q*@infinispan.org>
+          branch: pr_request
+          push-to-fork: infinispan-qe-bot/community-operators
+          title: 'Gingersnap Operator ${{ env.release-version }}'
```
[the remark this answers is not in the document]
Comment (pminz) [reply .github/workflows/image_release.yaml]: added

File: .github/workflows/release.yaml [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+          repository: k8s-operatorhub/community-operators
+          path: community-operators
+          ref: main
+
+      - name: Copy contents of bundle/ 
+        run: |
+          mkdir -p community-operators/operators/gingersnap/${{ env.release-version }}
+          cp -R bundle/* community-operators/operators/gingersnap/${{ env.release-version }}
+          git add community-operators/operators/gingersnap/${{ env.release-version }}
+          git commit -m "Operator Bundle ${{ env.release-version }}"
```
Comment (ryanemerson) [inline .github/workflows/release.yaml:79]: This commit will exist in the community-operators repo that is full of other operator bundles, therefore we should give this commit a more meaningful name.

How about: `"Gingersnap ${{ env.release-version }}"`?

File: .github/workflows/release.yaml [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+          repository: redhat-openshift-ecosystem/community-operators-prod
+          path: community-operators-prod
+          ref: main
+
+      - name: Copy contents of bundle/ and issue PRs
+        run: |
+          mkdir -p community-operators-prod/operators/gingersnap/${{ env.release-version }}
+          cp -R bundle/* community-operators-prod/operators/gingersnap/${{ env.release-version }}
+          git add community-operators-prod/operators/gingersnap/${{ env.release-version }}
+          git commit -m "Operator Bundle ${{ env.release-version }}"
```
Comment (ryanemerson) [inline .github/workflows/release.yaml:107]: How about: "Gingersnap ${{ env.release-version }}"?

File: .github/scripts/directory.sh [hunk 12]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
```
Comment (ryanemerson) [inline .github/scripts/directory.sh:1]: This file can be removed as no longer required.

Comment (pminz) [reply .github/scripts/directory.sh]: deleted

File: .github/workflows/release.yaml (hunk 11 above)
Comment (pminz) [reply .github/workflows/release.yaml]: yes . this commit message is better . updated

File: .github-scheduled-workflows/publish-main.yaml [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+
+jobs:
+  main:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Configure Git
+        run: |
+          git config --global user.name "pminz"
+          git config --global user.email "pminz@redhat.com"
```
Comment (ryanemerson) [inline .github-scheduled-workflows/publish-main.yaml:29]: We should use `infinispan-qe-bot` credentials here to be consistent.

**Comment by ryanemerson** [comment]
Thanks @pminz

**Pull request merged**